tests/integration/agenda/test_viewset.py

```diff
@@ -456,6 +450,79 @@ def test_mark_speaker_non_admin(self):
 
         self.assertEqual(response.status_code, 403)
 
+    # re-add last speaker
+    def util_add_user_as_last_speaker(self):
+        speaker = Speaker.objects.add(self.user, self.list_of_speakers)
+        speaker.begin_time = timezone.now()
+        speaker.end_time = timezone.now()
+        speaker.weight = None
+        speaker.save()
+
+    def test_readd_last_speaker_no_speaker(self):
+        response = self.client.post(
+            reverse(
+                "listofspeakers-readd-last-speaker", args=[self.list_of_speakers.pk]
+            )
+        )
+        self.assertEqual(response.status_code, 400)
+
+    def test_readd_last_speaker_no_last_speaker(self):
+        Speaker.objects.add(self.user, self.list_of_speakers)
+        response = self.client.post(
+            reverse(
+                "listofspeakers-readd-last-speaker", args=[self.list_of_speakers.pk]
+            )
+        )
+        self.assertEqual(response.status_code, 400)
+
+    def test_readd_last_speaker_has_last_speaker_no_next_speaker(self):
+        self.util_add_user_as_last_speaker()
+
+        response = self.client.post(
+            reverse(
+                "listofspeakers-readd-last-speaker", args=[self.list_of_speakers.pk]
+            )
+        )
+        self.assertEqual(response.status_code, 200)
+        speaker = Speaker.objects.get()
+        self.assertTrue(
+            speaker.begin_time is None
+            and speaker.end_time is None
+            and speaker.weight is not None
+        )
+
+    def test_readd_last_speaker_has_last_speaker_and_next_speaker(self):
+        self.util_add_user_as_last_speaker()
+        user2 = get_user_model().objects.create_user(
+            username="test_user_KLGHjkHJKBhjJHGGJKJn",
+            password="test_password_JHt678VbhjuGhj76hjGA",
+        )
+        Speaker.objects.add(user2, self.list_of_speakers)
+
+        response = self.client.post(
+            reverse(
+                "listofspeakers-readd-last-speaker", args=[self.list_of_speakers.pk]
+            )
+        )
+        self.assertEqual(response.status_code, 200)
+        speaker = Speaker.objects.get(pk=self.user.pk)
```

Comment:
Currently you try to get the spekaer with the spekaer-id of the user-id (these are two different pk-fields). But you got luck, that the id is correct (try to create user3 and use this one ;) ). -> `Speaker.objects.get(user__pk=self.user.pk)`